### ISSUE TITLE
Dahdi module installation failure

### ISSUE THREAD

**Issue by serg89** [ascom-au/asterisk-role#3]
The dahdi module in the debian repositories has recently been updated from v2.5.0 to v2.5.0.1 but this has caused the asterisk role to fail installation:

    TASK: [asterisk | dahdi-dkms patch] ******************************************* 
    <127.0.0.1> REMOTE_MODULE command patch -p0 -d / < roles/asterisk/files/patch.diff #USE_SHELL
    <127.0.0.1> EXEC ['/bin/sh', '-c', 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1429167641.97-23291215654144 && echo $HOME/.ansible/tmp/ansible-tmp-1429167641.97-23291215654144']
    <127.0.0.1> PUT /tmp/tmpkptAjQ TO /home/developer/.ansible/tmp/ansible-tmp-1429167641.97-23291215654144/command
    <127.0.0.1> EXEC ['/bin/sh', '-c', u'LANG=C LC_CTYPE=C /usr/bin/python /home/developer/.ansible/tmp/ansible-tmp-1429167641.97-23291215654144/command; rm -rf /home/developer/.ansible/tmp/ansible-tmp-1429167641.97-23291215654144/ >/dev/null 2>&1']
    failed: [127.0.0.1] => {"changed": true, "cmd": "patch -p0 -d / < roles/asterisk/files/patch.diff", "delta": "0:00:00.093645", "end": "2015-04-16 17:00:42.156860", "rc": 1, "start": "2015-04-16 17:00:42.063215"}
    stdout: can't find file to patch at input line 3
    Perhaps you used the wrong -p or --strip option?
    The text leading up to this was:
    --------------------------
    |--- /usr/src/dahdi-2.5.0.1+dfsg-1ubuntu3/drivers/dahdi/xpp/xbus-sysfs.c.ori    2014-06-04 23:18:24.644291060 +0200
    |+++ /usr/src/dahdi-2.5.0.1+dfsg-1ubuntu3/drivers/dahdi/xpp/xbus-sysfs.c        2014-06-04 23:22:30.945233860 +0200
    --------------------------
    File to patch: 
    Skip this patch? [y] 
    Skipping patch.
    2 out of 2 hunks ignored

    FATAL: all hosts have already failed -- aborting

After installation the dahdi module is not installed:

    Connected to Asterisk 11.7.0~dfsg-1ubuntu1 currently running on ENSUREspeech (pid = 7326)
    ENSUREspeech*CLI> dahdi show status
    No DAHDI found. Unable to open /dev/dahdi/ctl: No such file or directory
    Command 'dahdi show status' failed.

**Comment by serg89** [comment]
## Analysis

The problem appears to be that since dahdi has been updated the **dahdi-dkms patch** task fails since the file has moved. Any further tasks are not run and dahdi is never compiled and installted. Upon futher analysis it appears this patch is no longer needed anyway. But to keep consistency the best method is to allow the task to run but if the patch fails to ignore the error. In this way the patch only gets applied for the appropriate version of dahdi and all tasks which follow this are still run.

**Comment by zhousidun82** [comment]
## Review:
Code change looks good. Please merge to master.

**Comment by serg89** [comment]
## Review Response

Commit for issue #3 - https://github.com/ascom-au/asterisk-role/commit/769276cf7e54d931819e6e67b204452ee03cefd0 - merged branch ensurespeech-playbook#3 into master

**Comment by ncallister** [comment]
## QA

The DAHDI module now appears to install correctly:

```
Connected to Asterisk 11.7.0~dfsg-1ubuntu1 currently running on ENSUREspeech (pid = 778)
ENSUREspeech*CLI> dahdi show status
Description                              Alarms  IRQ    bpviol CRC    Fra Codi Options  LBO
DAHDI_DUMMY/1 (source: HRtimer) 1        UNCONFI 0      0      0      CAS Unk           0 db (CSU)/0-133 feet (DSX-1)
```
